### PR TITLE
Fix finite strain aniso

### PR DESCRIPTION
**Pull request by tduretz**
two fixes:
- clean up computations of `aniso_factor`
- add new preconditioner for linear solver (-1)

### CODE REVIEW

**Review by halterw — APPROVED** [review]

File: MDLIB/AnisotropyRoutines.c [hunk 1]
```diff
@@ -349,83 +349,126 @@ double ViscosityConciseAniso( int phase, double lxlz, double lx2, double angle,
...
-    
-    // Modify only in active cells
-    if ( mesh->BCp.type[c0] != 30 && mesh->BCp.type[c0] != 31 ) {
+      // Cell center index
+      c0 = k  + l*(Ncx);
 
-      // Loop on phases
-      for (int p=0; p<model->Nb_phases; p++) {
+      // First - initialize to 0
+      if ( model->anisotropy == 1 ) mesh->aniso_factor_n[c0] = 0.0;
```
Comment (kulakovri) [inline MDLIB/AnisotropyRoutines.c]: I saw multiple checks for anisotropy, but before calling the function you're checking it anyways. Do you need to add it here?

File: MDLIB/InputOutput.c [hunk 2]
```diff
@@ -1087,7 +1087,7 @@ Input ReadInputFile( char *fileName ) {
     model.penalty            = ReadDou2( fin, "penalty",           1.0e3 ); // Penalty factor
     model.auto_penalty       = ReadDou2( fin, "auto_penalty",        0.0 ); // Activates automatic penalty factor computation
     model.diag_scaling       = ReadInt2( fin, "diag_scaling",          1 ); // Activates diagonal scaling
-    model.preconditioner     = ReadInt2( fin, "preconditioner",        0 ); // Preconditoner type for Newton iterations, 0: Picard preconditionner
+    model.preconditioner     = ReadInt2( fin, "preconditioner",        0 ); // Preconditoner type for Linear solver, 0: Picard preconditionner, -1: symmetrised Picard, 2: symmetrised Newton
```
Comment (kulakovri) [inline MDLIB/InputOutput.c]: In the code I see only options -1, 0, 1, I don't see option 2.

File: MDLIB/Solvers.c [hunk 3]
```diff
@@ -1139,6 +1139,12 @@ void KillerSolver( SparseMat *matA,  SparseMat *matB,  SparseMat *matC,  SparseM
     //------------------------------------------------------------------------------------------------//
 
     // Construct preconditionner:
+    if ( preconditioner == -1 ) {
```
Comment (kulakovri) [inline MDLIB/Solvers.c]: Could you please comment different types of preconditioners here as well?

**Review by kulakovri — CHANGES_REQUESTED** [review]

File: MDLIB/InputOutput.c (hunk 2 above)
Comment (tduretz) [reply MDLIB/InputOutput.c]: correct, fixed

File: MDLIB/AnisotropyRoutines.c (hunk 1 above)
Comment (tduretz) [reply MDLIB/AnisotropyRoutines.c]: Right, removed superfluous checks

**Review by kulakovri — APPROVED** [review]

**Pull request merged**